mkpasswd doesn't cross compile

## Issue description

```
 9918 [void@fermat:~/pi]$ sh ./mkpasswd.sh
 9919 these derivations will be built:
 9920   /nix/store/f1jvccdwx8k1qdv2r88w9r3fz5fmynxg-mkpasswd-5.4.0-armv7l-unknown-linux-gnueabihf.drv
 9921 building '/nix/store/f1jvccdwx8k1qdv2r88w9r3fz5fmynxg-mkpasswd-5.4.0-armv7l-unknown-linux-gnueabihf.drv'...
 9922 unpacking sources
 9923 unpacking source archive /nix/store/k1psihqq886am72glfgh7q49pki9nn3k-source
 9924 source root is source
 9925 patching sources
 9926 updateAutotoolsGnuConfigScriptsPhase
 9927 configuring
 9928 no configure script, doing nothing
 9929 building
 9930 /bin/sh: pkg-config: not found
 9931 /bin/sh: pkg-config: not found
 9932 /bin/sh: pkg-config: not found
 9933 armv7l-unknown-linux-gnueabihf-gcc   -g -O2 -MM -MG *.c > Makefile.depend
 9934 /bin/sh: pkg-config: not found
 9935 /bin/sh: pkg-config: not found
 9936 /bin/sh: pkg-config: not found
 9937 perl make_version_h.pl debian/changelog > version.h
 9938 /bin/sh: perl: not found
 9939 make: *** [Makefile:85: version.h] Error 127
 9940 make: *** Deleting file 'version.h'
 9941 builder for '/nix/store/f1jvccdwx8k1qdv2r88w9r3fz5fmynxg-mkpasswd-5.4.0-armv7l-unknown-linux-gnueabihf.drv' failed with exit code 2
 9942 error: build of '/nix/store/f1jvccdwx8k1qdv2r88w9r3fz5fmynxg-mkpasswd-5.4.0-armv7l-unknown-linux-gnueabihf.drv' failed
 9943 
 9944 [void@fermat:~/pi]$ sh ./mkpasswd.sh
 9945 error: undefined variable 'pkgconfig' at /home/void/nixpkgs/pkgs/tools/security/mkpasswd/default.nix:8:25
 9946 (use '--show-trace' to show detailed location information)
 9947 
 9948 [void@fermat:~/pi]$ sh ./mkpasswd.sh
 9949 these derivations will be built:
 9950   /nix/store/df2lc9cr6x73xspy0jq9szj870jdk5pc-mkpasswd-5.4.0-armv7l-unknown-linux-gnueabihf.drv
 9951 building '/nix/store/df2lc9cr6x73xspy0jq9szj870jdk5pc-mkpasswd-5.4.0-armv7l-unknown-linux-gnueabihf.drv'...
 9952 unpacking sources
 9953 unpacking source archive /nix/store/k1psihqq886am72glfgh7q49pki9nn3k-source
 9954 source root is source
 9955 patching sources
 9956 updateAutotoolsGnuConfigScriptsPhase
 9957 configuring
 9958 no configure script, doing nothing
 9959 building
 9960 armv7l-unknown-linux-gnueabihf-gcc   -g -O2 -MM -MG *.c > Makefile.depend
 9961 perl make_version_h.pl debian/changelog > version.h
 9962 /bin/sh: perl: not found
 9963 make: *** [Makefile:85: version.h] Error 127
 9964 make: *** Deleting file 'version.h'
 9965 builder for '/nix/store/df2lc9cr6x73xspy0jq9szj870jdk5pc-mkpasswd-5.4.0-armv7l-unknown-linux-gnueabihf.drv' failed with exit code 2
 9966 error: build of '/nix/store/df2lc9cr6x73xspy0jq9szj870jdk5pc-mkpasswd-5.4.0-armv7l-unknown-linux-gnueabihf.drv' failed
 9967 
```

### Steps to reproduce
```
 9996 [void@fermat:~/pi]$ cat ./mkpasswd.sh
  9997 #!/bin/sh
  9998 nix-build -I nixpkgs=/home/void/nixpkgs -E 'let t1 = import <nixpkgs> {}; in let pkgs = import <nixpkgs> { crossSystem = t1.lib.systems.examples.armv7l-hf-multiplatform // { system = "armv7l-linux"; }; }; in pkgs.mkpasswd '
  9999 
```

Adding below fixs the issue.

```
  nativeBuildInputs = [ pkgconfig perl ];
```

--- a/pkgs/tools/security/mkpasswd/default.nix
+++ b/pkgs/tools/security/mkpasswd/default.nix
@@ -5,7 +5,7 @@ stdenv.mkDerivation {
 
   src = whois.src;
 
-  buildInputs = [ perl ];
+  nativeBuildInputs = [ perl ];
 
   preConfigure = whois.preConfigure;
   buildPhase = "make mkpasswd";

